### DISCUSSION TITLE
sdkver with NDK

### DISCUSSION THREAD

**Discussion by zebede22** [xmake-io/xmake-repo#3207] [Q&A]
In the libbpf example we have something like this:
`
add_requires("ndk >= 22")
set_toolchains("@ndk", {
    sdkver = "21"
})
`
`ndk` makes sense, it will fetch from xmake-repo.
what about `sdk` where it's coming from? and why `21`
does is coming from xmake-repo or is it part of ndk?

**Comment by waruqi** [comment]
just ndk platform api version, it still comes from ndk.

**Reply by waruqi** [reply]
it will define `-D__ANDROID_API__=21` when building android code.
